base/reducedim.jl

```diff
@@ -3,28 +3,41 @@
 ## Functions to compute the reduced shape
 
 # for reductions that expand 0 dims to 1
+reduced_index(i::OneTo) = OneTo(1)
+reduced_index(i::Slice) = Slice(first(i):first(i))
```

Comment:
It looks like simply making this a `UnitRange` instead of a `Slice` will do the trick — `make -C test reduce reduced offsetarray` passes for me locally with that change.